### ISSUE TITLE
Video capture orientation not handled proper

### ISSUE THREAD

**Issue by Aks-4125** [CrazyOrr/FFmpegRecorder#37]
I have been doing R & D on capturing video in square format and output for the same. Library does meet the requirements but the issue is whenever i capture video with portrait/landscape mode then resulting video is not rotating as per screen like native recorder.

![device-2018-01-04-144441](https://user-images.githubusercontent.com/29367708/34556810-f10acdee-f15d-11e7-92b9-198e6e229bea.png)

Please let me know what needs to be done here to overcome orientation problem. If I get success then I will do PR for contribution.

Device info : Moto G play Android 6.0

**Comment by CrazyOrr** [comment]
Check out the [transpose](https://ffmpeg.org/ffmpeg-filters.html#transpose) filter used in `VideoRecordThread`.

**Comment by Aks-4125** [comment]
@CrazyOrr  Thanks for quick response. transpose is already handled in `VideoRecordThread` as `cclock` for `270` degree rotation but resulting video is not rotating to `0` degree vertical view.  please help

**Comment by CrazyOrr** [comment]
In this demo I lock the screen orientation with `android:screenOrientation="portrait"` and only handle the case of transpose in portrait mode. 
If you wanna support landscape mode, you should remove the screen lock and try out the right way to transpose, also you may need to [swap width and height of preview](https://github.com/CrazyOrr/FFmpegRecorder/issues/31#issuecomment-342792295).

**Comment by Aks-4125** [comment]
@CrazyOrr Thanks for pointing out transpose filter. I am able to rotate video but facing issue in rotating as per documentation.

I tried to hardcode filter as following.

` FFmpegFrameFilter frameFilter = new FFmpegFrameFilter("transpose=cclock",
                    previewWidth, previewHeight);
            frameFilter.setPixelFormat(avutil.AV_PIX_FMT_NV21);
            frameFilter.setFrameRate(frameRate);`


According to project, `transpose=cclock` this filter should rotate video counter clockwise 90deg. but it is rotating video for 180 degree. 


I also tried to use `transpose=0/1/2/3`
`
0 = 90CounterCLockwise and Vertical Flip (default) (this does nothing)
1 = 90Clockwise (even this is also does nothing)
2 = 90CounterClockwise (this is rotating video at 180 degree)
3 = 90Clockwise and Vertical Flip (this behaves same as value2 : 120 degree rotation)
`



In short All i want, is rotate video 90 degree counter clockwise.. which I am not able to achieve. 

please help.

**Comment by CrazyOrr** [comment]
It's a little bit complicated, I have just made some changes and added support for landscape orientation in a223448387fed53fc7e7af24d56eed553d27051e. You can pull the latest code to see how it goes.

**Comment by Aks-4125** [comment]
@CrazyOrr Thanks for solution. working like charm. In case if you want to answer, that what you have done to resolve orientation issue. [Stack Overflow](https://stackoverflow.com/questions/48109876/javacv-transpose-ffmpegframefilter-is-not-working-properly)
